### ISSUE TITLE
[QUIC & HTTP/3] Handshake Timeout on tests

### ISSUE THREAD

**Issue by liveans** [dotnet/runtime#104426]
## Build Information
Build: https://dev.azure.com/dnceng-public/public/_build/results?buildId=719528
Build error leg or test failing: 

## Error Message

Fill the error message using [step by step known issues guidance](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssueJsonStepByStep.md).

```json
{
  "ErrorMessage": "Connection handshake was canceled due to the configured timeout of",
  "BuildRetry": false,
  "ExcludeConsoleLog": false
}
```

 ### Known issue validation
**Build: :mag_right:** https://dev.azure.com/dnceng-public/public/_build/results?buildId=719528
**Error message validated:** `[Connection handshake was canceled due to the configured timeout of`]
**Result validation:** :white_check_mark: Known issue matched with the provided build.
**Validation performed at:** 7/4/2024 11:05:15 AM UTC

### Report

|Build|Definition|Test|Pull Request|
|---|---|---|---|
|[1111741](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1111741)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1111741&view=ms.vss-test-web.build-test-results-tab&runId=30516512&resultId=223835)|dotnet/runtime#118060|
|[1111445](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1111445)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1111445&view=ms.vss-test-web.build-test-results-tab&runId=30507758&resultId=222196)|dotnet/runtime#118156|
|[1111352](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1111352)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1111352&view=ms.vss-test-web.build-test-results-tab&runId=30505636&resultId=223835)|dotnet/runtime#118207|
|[1110107](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1110107)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1110107&view=ms.vss-test-web.build-test-results-tab&runId=30462186&resultId=225286)|dotnet/runtime#118228|
|[1109871](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1109871)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1109871&view=ms.vss-test-web.build-test-results-tab&runId=30453746&resultId=225286)|dotnet/runtime#117635|
|[1109355](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1109355)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1109355&view=ms.vss-test-web.build-test-results-tab&runId=30439010&resultId=223803)|dotnet/runtime#114619|
|[1107642](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1107642)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1107642&view=ms.vss-test-web.build-test-results-tab&runId=30368192&resultId=223800)|dotnet/runtime#118132|
|[1107306](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1107306)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1107306&view=ms.vss-test-web.build-test-results-tab&runId=30355642&resultId=225525)|dotnet/runtime#118007|
|[1106649](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1106649)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1106649&view=ms.vss-test-web.build-test-results-tab&runId=30329788&resultId=225259)|dotnet/runtime#118079|
|[1105486](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1105486)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1105486&view=ms.vss-test-web.build-test-results-tab&runId=30293042&resultId=225248)|dotnet/runtime#117635|
|[1105339](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1105339)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1105339&view=ms.vss-test-web.build-test-results-tab&runId=30289012&resultId=225248)|dotnet/runtime#118037|
|[1103609](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1103609)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1103609&view=ms.vss-test-web.build-test-results-tab&runId=30225158&resultId=223785)|dotnet/runtime#117950|
|[1102265](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1102265)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1102265&view=ms.vss-test-web.build-test-results-tab&runId=30184488&resultId=225052)|dotnet/runtime#117797|
|[1101694](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1101694)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1101694&view=ms.vss-test-web.build-test-results-tab&runId=30158148&resultId=223336)|dotnet/runtime#117856|
|[1100488](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1100488)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1100488&view=ms.vss-test-web.build-test-results-tab&runId=30111132&resultId=223678)|dotnet/runtime#117593|
|[1100229](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1100229)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1100229&view=ms.vss-test-web.build-test-results-tab&runId=30103248&resultId=223678)|dotnet/runtime#117817|
|[1099361](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1099361)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1099361&view=ms.vss-test-web.build-test-results-tab&runId=30066874&resultId=223654)|dotnet/runtime#117821|
|[1098883](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1098883)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1098883&view=ms.vss-test-web.build-test-results-tab&runId=30054090&resultId=223560)|dotnet/runtime#117748|
|[1098834](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1098834)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1098834&view=ms.vss-test-web.build-test-results-tab&runId=30052944&resultId=221942)|dotnet/runtime#117277|
|[1098773](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1098773)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1098773&view=ms.vss-test-web.build-test-results-tab&runId=30050422&resultId=223560)|dotnet/runtime#117802|
|[1098738](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1098738)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1098738&view=ms.vss-test-web.build-test-results-tab&runId=30048732&resultId=223811)|dotnet/runtime#116798|
|[1098488](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1098488)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1098488&view=ms.vss-test-web.build-test-results-tab&runId=30036472&resultId=223554)|dotnet/runtime#117797|
|[1098403](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1098403)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1098403&view=ms.vss-test-web.build-test-results-tab&runId=30034244&resultId=224671)|dotnet/runtime#116947|
|[1098369](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1098369)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1098369&view=ms.vss-test-web.build-test-results-tab&runId=30033722&resultId=222093)|dotnet/runtime#117792|
|[1097037](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1097037)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1097037&view=ms.vss-test-web.build-test-results-tab&runId=29988158&resultId=222090)|dotnet/runtime#117729|
|[1096745](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1096745)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1096745&view=ms.vss-test-web.build-test-results-tab&runId=29978176&resultId=223551)|dotnet/runtime#117715|
|[1096445](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1096445)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1096445&view=ms.vss-test-web.build-test-results-tab&runId=29969500&resultId=223551)|dotnet/runtime#117720|
|[1095628](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1095628)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1095628&view=ms.vss-test-web.build-test-results-tab&runId=29961144&resultId=222090)|dotnet/runtime#117687|
|[1095831](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1095831)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1095831&view=ms.vss-test-web.build-test-results-tab&runId=29929418&resultId=223551)|dotnet/runtime#117398|
|[1094554](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1094554)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1094554&view=ms.vss-test-web.build-test-results-tab&runId=29858730&resultId=223551)|dotnet/runtime#117590|
|[1094337](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1094337)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1094337&view=ms.vss-test-web.build-test-results-tab&runId=29852892&resultId=223551)|dotnet/runtime#117629|
|[1094007](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1094007)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1094007&view=ms.vss-test-web.build-test-results-tab&runId=29844264&resultId=222090)|dotnet/runtime#117613|
|[1093412](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1093412)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1093412&view=ms.vss-test-web.build-test-results-tab&runId=29815494&resultId=222090)|dotnet/runtime#117578|
|[1092836](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1092836)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1092836&view=ms.vss-test-web.build-test-results-tab&runId=29796542&resultId=222089)|dotnet/runtime#117437|
|[1092246](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1092246)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1092246&view=ms.vss-test-web.build-test-results-tab&runId=29777244&resultId=222096)|dotnet/runtime#117540|
|[1092212](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1092212)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1092212&view=ms.vss-test-web.build-test-results-tab&runId=29776116&resultId=223540)|dotnet/runtime#110472|
|[1092193](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1092193)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1092193&view=ms.vss-test-web.build-test-results-tab&runId=29775296&resultId=222079)|dotnet/runtime#117465|
|[1091920](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1091920)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1091920&view=ms.vss-test-web.build-test-results-tab&runId=29758142&resultId=223592)|dotnet/runtime#117259|
|[1091832](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1091832)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1091832&view=ms.vss-test-web.build-test-results-tab&runId=29755252&resultId=220465)|dotnet/runtime#117532|
|[1091758](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1091758)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1091758&view=ms.vss-test-web.build-test-results-tab&runId=29752648&resultId=222080)|dotnet/runtime#117530|
|[1091177](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1091177)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1091177&view=ms.vss-test-web.build-test-results-tab&runId=29734022&resultId=222079)|dotnet/runtime#117507|
|[1091170](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1091170)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1091170&view=ms.vss-test-web.build-test-results-tab&runId=29733498&resultId=222079)|dotnet/runtime#116539|
|[1090911](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1090911)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1090911&view=ms.vss-test-web.build-test-results-tab&runId=29721342&resultId=222079)|dotnet/runtime#117460|
|[1090193](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1090193)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1090193&view=ms.vss-test-web.build-test-results-tab&runId=29698014&resultId=222077)|dotnet/runtime#117474|
|[1090153](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1090153)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1090153&view=ms.vss-test-web.build-test-results-tab&runId=29697930&resultId=222081)|dotnet/runtime#117473|
|[1089719](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1089719)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1089719&view=ms.vss-test-web.build-test-results-tab&runId=29681106&resultId=222071)|dotnet/runtime#117436|
|[1089076](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1089076)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1089076&view=ms.vss-test-web.build-test-results-tab&runId=29664598&resultId=223532)|dotnet/runtime#117447|
|[1088254](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1088254)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1088254&view=ms.vss-test-web.build-test-results-tab&runId=29640626&resultId=223532)|dotnet/runtime#117412|
|[1088042](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1088042)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1088042&view=ms.vss-test-web.build-test-results-tab&runId=29630458&resultId=222071)|dotnet/runtime#116772|
|[1087918](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1087918)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1087918&view=ms.vss-test-web.build-test-results-tab&runId=29624212&resultId=222072)|dotnet/runtime#117406|
|[1087534](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1087534)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1087534&view=ms.vss-test-web.build-test-results-tab&runId=29613784&resultId=223534)|dotnet/runtime#106374|
|[1086794](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1086794)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.CertificateCallbackThrowPropagates](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1086794&view=ms.vss-test-web.build-test-results-tab&runId=29584388&resultId=222071)|dotnet/runtime#116682|
#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|0|7|52|

**Comment by liveans** [comment]
This issue is not directly related to #99074 (that is specific test, this is more general issue), so it's not a duplicate.

**Comment by ManickaP** [comment]
Looks like this is a instance of https://github.com/microsoft/msquic/issues/4132. On OSX, We're using MsQuic at this commit: https://github.com/microsoft/msquic/commit/e0385b047d7bd622319e9f6c4c00184a1294eea8 build from main (therefore having version of the next release - 2.4). 
Historically, the above mentioned issue had a fix by that time: https://github.com/microsoft/msquic/pull/4145 
Unfortunately, the were at least 2 follow ups it seems: 
- https://github.com/microsoft/msquic/pull/4169
- https://github.com/microsoft/msquic/pull/4245

Which are not part of the binaries with which we're testing.
Also this is likely the reason for all the other timeouts and hangs on OSX.

TL;DR This should get fixed by new MsQuic, i.e. with [#114912](https://github.com/dotnet/runtime/issues/114912)

Alternatively, we can change the async cert validation condition to `>` in here: https://github.com/dotnet/runtime/blob/02596ba8d937bfbda5d2d7d634d7b3688e28bb69/src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicApi.cs#L66
